packages/neon/neon_notifications/lib/pages/main.dart

```diff
@@ -24,33 +24,39 @@ class _NotificationsMainPageState extends State<NotificationsMainPage> {
   }
 
   @override
-  Widget build(final BuildContext context) => ResultBuilder<List<NotificationsNotification>>.behaviorSubject(
-        stream: bloc.notifications,
-        builder: (final context, final notifications) => Scaffold(
-          resizeToAvoidBottomInset: false,
-          floatingActionButton: FloatingActionButton(
-            onPressed: () async {
-              bloc.deleteAllNotifications();
-            },
-            tooltip: AppLocalizations.of(context).notificationsDismissAll,
-            child: const Icon(MdiIcons.checkAll),
-          ),
-          body: NeonListView<NotificationsNotification>(
-            scrollKey: 'notifications-notifications',
-            withFloatingActionButton: true,
-            items: notifications.data,
-            isLoading: notifications.isLoading,
-            error: notifications.error,
-            onRefresh: bloc.refresh,
-            builder: _buildNotification,
-          ),
+  Widget build(final BuildContext context) {
+    final notificationCount = bloc.notifications.value.data?.length ?? 0; // Calculate the notification count
+
+    return ResultBuilder<List<NotificationsNotification>>.behaviorSubject(
+      stream: bloc.notifications,
+      builder: (final context, final notifications) => Scaffold(
+        resizeToAvoidBottomInset: false,
+        floatingActionButton: FloatingActionButton(
+          onPressed: notificationCount > 0 ? () async {
+            bloc.deleteAllNotifications();
+          } : null, // Disable the button if there are no notifications
```

Comment:
A lambda function here would be more readable.

Reply:
I'd prefer a conditional tearoff. There isn't any need to make this function asynchronous. 
```dart
FloatingActionButton(
  onPressed: notificationCount > 0
    ? bloc.deleteAllNotifications
    : null,
),
```

Reply:
Ah that is even better!